packages/neon/neon/lib/src/widgets/image.dart

```diff
@@ -192,42 +236,128 @@ class _NeonCachedImageState extends State<NeonCachedImage> {
       );
 }
 
+/// Nextcloud api image.
+///
+/// Extension for [NeonCachedImage] providing a [NextcloudClient] to the caller
+/// to retrieve the image.
+///
+/// See:
+///  * [NeonCachedImage] for a customized image
+///  * [NeonImageWrapper] for a wrapping widget for images
 class NeonApiImage extends StatelessWidget {
+  /// creates a new Neon api image.
   const NeonApiImage({
     required this.getImage,
     required this.cacheKey,
+    this.account,
     this.reviver,
     this.writeCache,
     this.isSvgHint = false,
     this.size,
     this.fit,
-    this.svgColor,
     this.iconColor,
     this.errorBuilder,
     super.key,
   });
 
+  /// Optional account to use for the request.
+  ///
+  /// Defaults to the currently active account in [AccountsBloc.activeAccount].
+  final Account? account;
+
+  /// Image downloader.
   final ApiImageDownloader getImage;
+
+  /// Cache key used for [NeonCachedImage.key].
   final String cacheKey;
+
+  /// Custom cache reviver function.
   final CacheReviver? reviver;
+
+  /// Custom cache writer function.
   final CacheWriter? writeCache;
+
+  /// {@macro NeonCachedImage.svgHint}
   final bool isSvgHint;
+
+  /// {@macro NeonCachedImage.size}
   final Size? size;
+
+  /// {@macro NeonCachedImage.fit}
   final BoxFit? fit;
-  final Color? svgColor;
+
+  /// {@macro NeonCachedImage.iconColor}
   final Color? iconColor;
+
+  /// {@macro NeonCachedImage.errorBuilder}
   final ErrorWidgetBuilder? errorBuilder;
 
   @override
   Widget build(final BuildContext context) {
-    final account = NeonProvider.of<AccountsBloc>(context).activeAccount.value!;
+    final account = this.account ?? NeonProvider.of<AccountsBloc>(context).activeAccount.value!;
```

Comment:
This feels quite dangerous because we should always load the images with the correct account. Can you mark the account field as non-nullable and required and then explicitly specify it this way everywhere where it is used?

Reply:
Not really

The old way did use the active one by default and this is needed as we (eventually) don't want to export the `AccountsBloc`.
So how would other clients then be able to use the currently active one.

Reply:
Either you always use the active account and note that down as the expected behavior or you need to always specify it. This way of having both options is dangerous in my opinion. Better to just hide the account field then

Reply:
I totally agree.
Do you think a separate constructor would help? Like `NeonApiImage.custom` that would require the account.

`NeonUserAvatar` already uses this option and while it has access to the `AccountsBloc` other ones like the potential `NeonTalk` client wouldn't

Reply:
Yes that seems to be the best option :thinking: